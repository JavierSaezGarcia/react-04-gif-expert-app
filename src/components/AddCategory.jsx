import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = (event) => {    // target es la desestructuracion de event
    setInputValue(event.target.value);
  }

  const onSubmit = (event) => {

    event.preventDefault(); // evito la recarga del navegador por el submit
    if (inputValue.trim().length <= 1) return;   // si hay menos de dos caracteres no hago nada
    
    onNewCategory(inputValue.trim()); // Le paso del hijo que es addCategory el valor al la funcion del padre onAddCategory
    setInputValue(''); // VacÃ­o el set del useState

  }
  return (
    
    <div className="col-xs-12 col-md-4">

      <form onSubmit={onSubmit}>
        <div className="input-group input-group-md mb-3">
          <input type="text"
            className="form-control"
            placeholder="Search Gifs"
            value={inputValue}
            onChange={onInputChange} />
          <button className="btn btn-primary" type="submit" onClick={onSubmit}>Search</button>
        </div>
      </form>

    </div>

  )
}
