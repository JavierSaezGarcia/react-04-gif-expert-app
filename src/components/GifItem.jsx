import PropTypes from 'prop-types';

export const GifItem = ({id, title, url}) => {
    const titulo= title.trim();
  return (
    <div className="card" key={id}>
        <div key={id}>                    
        <p>{titulo.toUpperCase()}</p>
        <img src={url} alt={title} />
        </div>
    </div>    
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}


/*  Tarea:
1. AÃ±adir proptypes a GifItem ??? yarn add Pro

  a. title obligatorio
  b. url obligatorio
2. Hacer y Evaluar el snapshot

*/