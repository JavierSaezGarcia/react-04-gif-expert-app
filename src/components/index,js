// Archivo barril para evitar monton de importaciones
// No funciona
// export * from './AddCategory';
// export * from './GifGrid';
// export * from './GifItem';

