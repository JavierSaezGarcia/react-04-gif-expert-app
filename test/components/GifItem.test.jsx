import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Testing in <GifItem />', () => {

  const title = 'soldiers amulet';
  const url = 'https://media4.giphy.com/media/Y1qPAAj8s5HuWqvPnw/giphy-downsized-medium.gif?cid=2cc56eb3fuelhqdvb1doohv4gmfb3mt38djcot1lkgq9gugd&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g';
  
  
  test('should match with the snapshot', () => {
    
    const { container } = render( <GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();

  });
  test('should show the image with the url and alt', () => {

    render( <GifItem title={title} url={url} />);  
    // screen.debug(); // Imprime información del DOM renderizado en la consola
    // TODO Manera menos bonita y eficiente:
    // expect(screen.getByRole('img').src)
    //       .toBe(url);
    // expect(screen.getByRole('img').alt)
    //       .toBe(title);

    // TODO Manera bonita y eficiente:
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

    // console.log('url imagen: ',screen.getByRole('img').src);
    // console.log('texto imagen: ',screen.getByRole('img').alt);
  });

  test('should display the title in the component', () => {

    render( <GifItem title={title} url={url} />);
    expect(screen.getByText(title.toUpperCase().trim())).toBeTruthy();
    
  });
  
});




/*  Tarea:
1. Añadir proptypes a GifItem ??? yarn add Pro

  a. title obligatorio
  b. url obligatorio
2. Hacer y Evaluar el snapshot

*/