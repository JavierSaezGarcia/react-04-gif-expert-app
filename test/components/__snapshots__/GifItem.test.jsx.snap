// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div>
      <p>
        SOLDIERS AMULET
      </p>
      <img
        alt="soldiers amulet"
        src="https://media4.giphy.com/media/Y1qPAAj8s5HuWqvPnw/giphy-downsized-medium.gif?cid=2cc56eb3fuelhqdvb1doohv4gmfb3mt38djcot1lkgq9gugd&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g"
      />
    </div>
  </div>
</div>
`;
